stdlib/public/SwiftShims/Visibility.h

```diff
@@ -101,41 +101,57 @@
 #if defined(__MACH__) || defined(__wasi__)
 
 # define SWIFT_EXPORT_ATTRIBUTE __attribute__((__visibility__("default")))
+# define SWIFT_IMPORT_ATTRIBUTE __attribute__((__visibility__("default")))
 
 #elif defined(__ELF__)
 
 // We make assumptions that the runtime and standard library can refer to each
 // other's symbols as DSO-local, which means we can't allow the dynamic linker
 // to relocate these symbols. We must give them protected visibility while
 // building the standard library and runtime.
-# if defined(swiftCore_EXPORTS)
-#  define SWIFT_EXPORT_ATTRIBUTE __attribute__((__visibility__("protected")))
-# else
-#  define SWIFT_EXPORT_ATTRIBUTE __attribute__((__visibility__("default")))
-# endif
+# define SWIFT_EXPORT_ATTRIBUTE __attribute__((__visibility__("protected")))
+# define SWIFT_IMPORT_ATTRIBUTE __attribute__((__visibility__("default")))
```

Comment:
This definitely breaks the uniformity with Windows.  Declarations for the module *must* use `dllexport` if they are for the current module (and exported), but use `dllimport` if they are being accessed outside of the module.  That is the reason for the `swiftCore_EXPORTS` macro and the conditional definition.

Reply:
I understand all that; I think you're misunderstanding what this bit of the patch is doing.  The existing code was doing two things at once: (1) stating the target-specific attributes that should be used for imported and exported declarations and (2) picking which attribute to use on declarations that expected to be exported by swiftCore (i.e. the attribute that ends up in `SWIFT_RUNTIME_EXPORT`).  I'm just separating those two decisions: first we figure out what attributes should be used for exports and imports generally, and then we decide which attribute to put in `SWIFT_RUNTIME_EXPORT` based on whether we're in swiftCore.  That lets us re-use the first decision when we have multiple runtime libraries, which we do with Differentiation and now with Concurrency.